### PR TITLE
TMHASH is 32 bytes. Closes #1990

### LINKED ISSUE #1990: Analyze cost to attack truncated SHA256
There is concern that 80 bits of security with truncated 20 byte sha256 is insufficient. We need to do a cost analysis of the cost to break truncated levels of sha256 (i.e. cost per asic + electricity) and figure out exactly what margin between 20 byte and 32 byte hashes we want to use. (As the space savings are worth considering) Its important to note that you can't reuse bitcoin asics, since we have domain seperators on the hashes.

Ideally at the end of it, we should create an ADR for this.

### PR DESCRIPTION
**Pull request by ebuchman**
Closes #1990 

Change TMHASH to 32-bytes.

Include TMHASHTruncated as TMHASH[:20]

Add `crypto.AddressSize = TMHASHTruncated.Size`

* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: crypto/tmhash/hash.go [hunk 1]
```diff
@@ -6,10 +6,27 @@ import (
...
 )
 
+// New returns a new hash.Hash.
+func New() hash.Hash {
+	return sha256.New()
+}
+
+// Sum returns the SHA256 of the bz.
+func Sum(bz []byte) []byte {
+	h := sha256.Sum256(bz)
```
[the remark this answers is not in the document]
Comment (ebuchman) [reply crypto/tmhash/hash.go]: Can't because it's a `[32]byte` and not `[]byte`

[the remark this answers is not in the document]
Comment (Liamsi) [reply crypto/tmhash/hash.go]: Oh, I thought sha256 implements the hash.Hash interface which returns `[]byte`. I actually does but not via the "static" methods, only via `sha256.New().Sum(bz)`. Not important though!

[the remark this answers is not in the document]
Comment (alexanderbez) [reply crypto/tmhash/hash.go]: Could just do `sha256.Sum(bz)`[:], no?

[the remark this answers is not in the document]
Comment (ValarDragon) [reply crypto/tmhash/hash.go]: No, theres a weird reason why that doesn't work, I'm forgetting what it was.

**Pull request merged**